src/components/ProgressBar.tsx

```diff
@@ -0,0 +1,94 @@
+import classNames from "classnames";
+import { useState } from "react";
+
+type ProgressBarType = {
+  minValue?: number;
+  maxValue?: number;
+  rounded?: "none" | "md" | "full";
+  color?: "violet" | "pink" | "red" | "orange" | "yellow" | "lime" | "cyan";
+  disabled?: boolean;
+  showPercentage?: boolean;
+  className?: string;
+  currentValue?: number;
+};
+
+const ProgressBar = ({
+  minValue = 0,
+  maxValue = 100,
+  rounded = "none",
+  color = "cyan",
+  currentValue = 0,
+  showPercentage = true,
+  disabled,
+  className,
+}: ProgressBarType) => {
+
+  const clampedValue = Math.min(maxValue, Math.max(currentValue, minValue));
+  const widthPercentage = ((clampedValue - minValue) / (maxValue - minValue)) * 100;
+
+  return (
+    <div
+      className={classNames(
+        "w-72 md:w-full max-w-md border-black border-2 focus:outline-none h-9 overflow-hidden",
```

Comment:
Do you mind adding `shadow-[2px_2px_0px_rgba(0,0,0,1)] `to this style? I prefer the style with the shadow when a user doesn't need to do any interaction.

```suggestion
        "w-72 md:w-full max-w-md border-black border-2 focus:outline-none h-9 overflow-hidden shadow-[2px_2px_0px_rgba(0,0,0,1)]",
```